test/optimizations/autoLocalAccess/views.good

```diff
@@ -0,0 +1,83 @@
+
+Start analyzing forall (views.chpl:40)
+| Found loop domain (views.chpl:30)
+| Will attempt static and dynamic optimizations (views.chpl:40)
+|
+|  Start analyzing call (views.chpl:41)
+|   Found the domain of the access base (views.chpl:29)
+|  This call is a dynamic optimization candidate (views.chpl:41)
+|
+|  Start analyzing call (views.chpl:42)
+|   Can't determine the domain of access base (views.chpl:35)
+|  This call is a dynamic optimization candidate (views.chpl:42)
+|
+End analyzing forall (views.chpl:40)
+
+
+Start analyzing forall (views.chpl:61)
+| Found loop domain (views.chpl:53)
+| Will attempt static and dynamic optimizations (views.chpl:61)
+|
+|  Start analyzing call (views.chpl:63)
+|   Can optimize: Access base is the iterator's base (views.chpl:63)
+|  This call is a static optimization candidate (views.chpl:63)
+|
+|  Start analyzing call (views.chpl:64)
+|   Can't determine the domain of access base (views.chpl:56)
+|  This call is a dynamic optimization candidate (views.chpl:64)
+|
+End analyzing forall (views.chpl:61)
+
+
+Start analyzing forall (views.chpl:84)
+| Found loop domain (views.chpl:80)
+| Will attempt static and dynamic optimizations (views.chpl:84)
+|
+|  Start analyzing call (views.chpl:86)
+|   Can optimize: Access base is the iterator's base (views.chpl:86)
```

Comment:
I haven't really studied this output in much detail before.  Is the +offset array access skipped in this analysis?  (it doesn't seem to be mentioned).  And it doesn't seem to either force dynamic optimization nor diable the optimization?

Reply:
Yes, it is skipped, there is no static or dynamic optimization for that case. We give up when we see a non-SymExpr as the index right-away. So `arr[i+0]` wouldn't be optimized for example.

Right before the last release I did some improvements that made this output more verbose for cases where it didn't optimize, but that case is still not covered apparently. I don't think it is a big effort to add that particular one.

Currently, if something is optimized it is reported for sure. But if we shy away from optimization, I wouldn't be surprised if we miss some other cases, too. I have added a checkbox for that under https://github.com/Cray/chapel-private/issues/1058

Reply:
OK, it's fine with me to leave it as-is for now.  I'd expected to see some sort of "skipping this expression" line (and think this would be more useful to an end-user ultimately), but don't have an objection to leaving it as-is for now.  (For it to truly be useful to an end-user, we'll probably also need to identify the expressions more directly than by line number since a typical forall loop body will have many expressions per line...).

Reply:
> For it to truly be useful to an end-user, we'll probably also need to identify the expressions more directly than by line number since a typical forall loop body will have many expressions per line...

I agree. I'd really want this ability to be through tracking column number (https://github.com/chapel-lang/chapel/issues/9027) similarly to how we track the line number, rather than trying to rebuild the expression.